### ISSUE TITLE
RFE: Automatic controller configurations.

### ISSUE THREAD

**Issue by i30817** [libretro/RetroArch#5064]
So as a small brainstorming exercise, i'd like the RA people to discuss how to make controller configurations more intuitive and approaching zero effort.

As RA keeps getting more and more systems, more and more per system controller systems are available for the games, some of which support 'X controller' some which don't, some controllers who are mappable to a native controller without user intervention, some which aren't.

I was thinking of how to make this better for users, and my thinking is that this pretty much requires a tag database. Now i'm pretty sure this would be great for filtering the verified scanned playlists, but this RFE is slightly more ambitious. I'd very much like that RE got to the point where a tag of 'this wii game supports classical controller' could interact with the dolphin core and a user option of 'preferred native controller order'

This would work, like say ordered lists on the core(s) with possibilities. For wii:

one list with these, that the user could modify the order
-- emulated wiimote, real wiimote, hybrid wiimote

another list with the accessory extension preference:
classical, nunchuck, guitar, drums, turntable

the order is for the user to express a *preference* in the case the game support multiple. Then when starting a game the frontend would check the core order preferences, use the database tag to filter out the relevant controllers for that game, and build one with the first 'match' on the preferences, in this case, emulated wiimote with classical extension, because the game 'supports' both.

This is relevant for all systems basically. Certainly psx that had games with mouse, lightguns and mats etc.

**Comment by hizzlekizzle** [comment]
I don't think I understand how this would actually work and interact with the existing paradigms. I guess you're proposing that we should be able to identify when a controller is an original controller and then pass that through directly to the core without running it through the input interface?

We could add entries to the core info files that list the controllers, but to what end? How does the frontend know what any of that means?

**Comment by i30817** [comment]
>  I guess you're proposing that we should be able to identify when a controller is an original controller and then pass that through directly to the core without running it through the input interface?

Not at all. The core of the idea is just to allow users to express preference in native controllers for games with multiple native controller support. Say, a game like policenauts on the ps1 can use the controllers OR ps1 mouse. This idea would allow a core to select mouse automatically without the user specifying it for that single game.

It requires a online database of tags associated to games, which would be useful for other things too, such as gamelist filtering.
